crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -110,9 +129,18 @@ impl ProtocolFees {
             buy: quote.buy_amount,
             fee: quote.fee,
         };
-        let protocol_fees = protocol_fees
+        let apply_protocol_fee = if let Ok(Some(address)) = order
+            .signature
+            .recover(domain_separator, &order.data.hash_struct())
+        {
+            self.driver_must_apply_protocol_fee(address.signer)
+        } else {
+            false
```

Comment:
I think the better default is to apply the fee when in doubt (although theoretically we should never enter this case anyway).